src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from 'react';
+
+import * as RadixCheckbox from '@radix-ui/react-checkbox';
+import { IconCheckBoxCheck } from '../Icons';
+
+import classNames from 'classnames';
+import styles from './Checkbox.module.scss';
+const cx = classNames.bind(styles);
+
+export interface CheckboxProps {
+  className?: string;
+  name: string;
+  text?: string;
+  link?: string;
+  isDisabled?: boolean;
+  checked?: boolean;
+  hasLightBackground?: boolean;
+  onChange?: (value: boolean) => void;
+  onLinkClick?: () => void;
+}
+
+export const Checkbox = ({
+  className,
+  name,
+  text,
+  link,
+  isDisabled,
+  checked,
+  hasLightBackground,
+  onChange,
+  onLinkClick
+}: CheckboxProps) => {
+  const [isChecked, setIsChecked] = useState(checked);
+
+  const handleChange = (checkState: boolean) => {
+    if (!isDisabled) {
+      setIsChecked(checkState);
+      onChange(checkState);
+    }
+  };
+
+  const handleClick = (event: React.MouseEvent<HTMLSpanElement>) => {
+    if (onLinkClick && !isDisabled) {
+      event.preventDefault();
```

Comment:
Changed to the div element. As we integrate the link inside the label to prevent the check from unchecking on the link click we have to add prevent default.